frontend/src/pages/Billing/BillingPageV2.tsx

```diff
@@ -104,13 +106,20 @@ const UsageCard = ({
 		start: moment.Moment
 		end: moment.Moment
 	}>({
-		start: (billingPeriodEnd ?? moment()).subtract(3, 'months'),
-		end: billingPeriodEnd ?? moment(),
+		start: (moment(billingPeriodEnd) ?? moment()).subtract(3, 'months'),
+		end: moment(billingPeriodEnd) ?? moment(),
 	})
 
-	const setUsageRange = (option: 'Monthly' | 'Weekly') => {
+	const setUsageRange = (option: 'Monthly' | 'Weekly' | 'Daily') => {
 		setRange({
-			start: moment().subtract(option === 'Monthly' ? 12 : 3, 'months'),
+			start: moment().subtract(
+				option === 'Monthly'
+					? 4 * 12 * 7
+					: option === 'Weekly'
+					? 12 * 7
+					: 12,
```

Comment:
Nit: Maybe for a follow up but may make it more readable to bring this 12 out into a constant? Was trying to figure out what this 12 means - concluded that we want to show 12 periods of which ever option they select